vaadin-upload-flow-parent/vaadin-upload-testbench/src/main/java/com/vaadin/flow/component/upload/testbench/UploadElement.java

```diff
@@ -151,10 +151,13 @@ private void waitForUploads(int maxSeconds) {
         timeouts.setScriptTimeout(maxSeconds, TimeUnit.SECONDS);
 
         String script = "var callback = arguments[arguments.length - 1];"
```

Comment:
Could we avoid using setInterval and implement this with waitUtil? My concern is that setInterval can run forever in the background if the web component gets stuck in the inProgress state.

Reply:
Make sense. Let me make a try

Reply:
BTW, is it OK to use Java multiline strings or should the change be compatible with older Java versions?

Reply:
done

Reply:
Thanks. I'll update the PR